File: models/photo.go
package models

type Photo struct {
	ID           int     `json:"id"`
	FileDir      string  `json:"file_dir"`
	Name         string  `json:"name"`
	CameraMake   string  `json:"camera_make"`
	CameraModel  string  `json:"camera_model"`
	LensID       string  `json:"lens_id"`
	Width        int     `json:"width"`
	Height       int     `json:"height"`
	FocalLength  float64 `json:"focal_length"`
	Aperture     float64 `json:"aperture"`
	ShutterSpeed string  `json:"shutter_speed"` // string to handle values in the form of a/b in database
	ISO          int     `json:"iso"`
	CapturedAt   string  `json:"captured_at"`
}

File: frontend/src/types/Photo.ts
export interface IPhoto {
    id: number
    name: string
    file_dir: string
    camera_make: string
    camera_model: string
    lens_id: string
    width: number
    adjusted_width: number
    height: number
    adjusted_height: number
    focal_length: number
    aperture: number
    shutter_speed: string
    iso: number
    captured_at: Date
}
